### PR TITLE
Add options dry-run and header-stamp

### PR DESCRIPTION
**Pull request by matks**
I have added a `Reporter` class in charge of monitoring the actions being done and 2 feature flags: `display-report` and `dry-run`

The display-report option will output something like this:
```
Header Stamp Report
-------------------

 Fixed:
 * module.js
 * webpack.config.js
 * module.js
 * module.css
 * navbar.tpl
 * end.tpl
 * welcome.tpl
 * content.tpl
 * tooltip.tpl
 * lost.tpl
 * popup.tpl
 * package.json
 * composer.json

 Nothing to fix:
 * AdminWelcomeController.php
 * Configuration.php
 * OnBoarding.php
 * welcome.php
 * OnBoarding.js
```

### CODE REVIEW

File: src/Reporter.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * 2007-2020 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
```
[the remark this answers is not in the document]
Comment (matks) [reply src/Reporter.php]: Fixed in https://github.com/PrestaShopCorp/header-stamp/pull/14/commits/caf50aa3b65ac7aa277298ab392f5e83aba88cf7

File: src/Command/UpdateLicensesCommand.php [hunk 2]
```diff
@@ -1,13 +1,13 @@
 <?php
 /**
- * 2007-2019 PrestaShop and Contributors
+ * 2007-2020 PrestaShop and Contributors
  *
  * NOTICE OF LICENSE
  *
- * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * This source file is subject to the Open Software License (OSL 3.0)
```
[the remark this answers is not in the document]
Comment (matks) [reply src/Command/UpdateLicensesCommand.php]: Fixed in https://github.com/PrestaShopCorp/header-stamp/pull/14/commits/caf50aa3b65ac7aa277298ab392f5e83aba88cf7

**Comment by matks** [comment]
Updated the PR !
- license headers of header-stamp are fixed
- rebased the PR to remove git conflicts
- improved the display of dry-run mode for better understanding of users

This is how it looks like in a GitHub Action:
Capture d’écran 2020-06-13 à 23 15 18

GitHub Action template:
```
name: PHP tests
on: [push, pull_request]
jobs:

  header-stamp:
    name: Header stamp
    runs-on: ubuntu-latest
    steps:
      - name: Setup PHP
        uses: shivammathur/setup-php@v1
        with:
          php-version: '7.3'
          extensions: xml, json
      - uses: actions/checkout@v1
        with:
          fetch-depth: 0

      - run: composer install --prefer-dist
      - run: ./vendor/bin/header-stamp --exclude=vendor,node_modules --dry-run
```

File: src/Reporter.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+ */
+
+namespace PrestaShop\HeaderStamp;
+
+/**
+ * Reporter in charge of reporting what HeaderStamp has done
+ */
+class Reporter
+{
+    private $report = [
```
Comment (PierreRambaud) [inline src/Reporter.php:34]:
```suggestion
    private $reports = [
```

There are reports :thinking:

Comment (matks) [reply src/Reporter.php:34]: It's a very big report 😅

**Comment by matks** [comment]
Thanks for review ❤️

**Pull request merged**